manifests/peers.pp

```diff
@@ -1,11 +1,25 @@
+#
 define haproxy::peers (
   $collect_exported = true,
+  $instance = 'haproxy',
 ) {
 
+  # We derive these settings so that the caller only has to specify $instance.
+  include haproxy::params
+  if $instance == 'haproxy' {
+    $instance_name = 'haproxy'
+    #$config_dir = $haproxy::params::config_dir
+    $config_file = $haproxy::params::config_file
+  } else {
+    $instance_name = "haproxy-${instance}"
+    #$config_dir = inline_template($haproxy::params::config_dir_tmpl)
+    $config_file = inline_template($haproxy::params::config_file_tmpl)
+  }
```

Comment:
I don't like repeating the same 10 lines in multiple files.  I couldn't come up with any other way to do this.

It can't be part of haproxy::params.pp because the value of $instance changes. 

If someone suggests a better way to do it, I'm willing to try anything!